pkg/storage/sasNode.go

```diff
@@ -80,6 +83,8 @@ func (sas *sasStorage) NodePublishVolume(ctx context.Context, req *csi.NodePubli
 	}
 	klog.Infof("attached device at %s", path)
 
+	checkPreviouslyRemovedDevices(ctx)
```

Comment:
Before you call checkPreviouslyRemovedDevices(), you should make sure that the current volume's wwn is removed from globalRemovedDevicesMap so you can handle the case where a volume is being re-mounted after previously being unmounted.